src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetServiceImpl.java

```diff
@@ -26,34 +31,53 @@
   private final DatasetRepository datasetRepository;
   private final AsyncDatasetPublishService publishService;
 
+  private final TransformationService transformationService;
+
   public DatasetServiceImpl(
       DatasetGeneratorService generatorService,
       DatasetRepository datasetRepository,
-      AsyncDatasetPublishService publishService) {
+      AsyncDatasetPublishService publishService,
+      TransformationService transformationService) {
     this.generatorService = generatorService;
     this.datasetRepository = datasetRepository;
     this.publishService = publishService;
+    this.transformationService = transformationService;
   }
 
   @Transactional
   @Override
   public Dataset createDataset(String datasetName, Country country, Language language,
-      List<ByteArrayInputStream> records) {
+      List<ByteArrayInputStream> records, InputStream xsltTransformerEDMExternal) {
```

Comment:
This is not a transformer but the contents of the xslt file and in fact an input stream. Perhaps name accordingly?